### PR TITLE
Make verifier "?" output similar to goal form

### PR DESCRIPTION
**Pull request by kamahen**
- Remove EVar(0x... = ...) wrapper
- Change App(vname, (...)) to vname(...)
- Change EVar(&lt;nullptr&gt;) to &lt;null&gt;
- Quote strings that don't conform to verifier ID syntax.

### CODE REVIEW

File: kythe/cxx/verifier/assertion_ast.h [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #include <algorithm>
 #include <unordered_map>
 #include <vector>
+#include <ctype.h>
+#include <regex>
```
Comment (shahms) [inline kythe/cxx/verifier/assertion_ast.h]: Please use RE2.  We already depend on it and the last time I checked all of the implementations of the standard header are terrible. The RE2 header is "re2/re2.h" and the build dep "@com_googlesource_code_re2//:re2"

File: kythe/cxx/verifier/assertion_ast.h [hunk 2]
```diff
@@ -78,7 +104,9 @@ class SymbolTable {
   /// Maps `Symbol`s back to their original text.
   std::vector<const std::string*> reverse_map_;
   /// The text to use for unique() symbols.
-  std::string unique_symbol_ = "(unique)";
+  const std::string unique_symbol_;
+  /// Used for quoting strings - see assertions.lex:
+  const std::regex id_regex_;
```
Comment (shahms) [inline kythe/cxx/verifier/assertion_ast.h]: Making these const disables efficient move semantics and all assignment; please remove the `const` qualifier.

File: kythe/cxx/verifier/assertion_ast.h [hunk 3]
```diff
@@ -35,6 +37,10 @@ typedef size_t Symbol;
 /// \brief Maps strings to `Symbol`s.
 class SymbolTable {
  public:
+  SymbolTable() :
```
Comment (shahms) [inline kythe/cxx/verifier/assertion_ast.h]: `explicit`

File: kythe/cxx/verifier/assertion_ast.h [hunk 4]
```diff
@@ -58,10 +64,30 @@ class SymbolTable {
     auto* text = reverse_map_[symbol];
     if (text == &unique_symbol_) {
       return "(unique#" + std::to_string(symbol) + ")";
-    } else if (!text->empty()) {
+    } else if (!text->empty() && regex_match(*text, id_regex_)) {
       return *text;
     } else {
-      return "\"\"";
+      std::string quoted("\"");
```
Comment (shahms) [inline kythe/cxx/verifier/assertion_ast.h]: `return absl::StrCat("\"", absl::CHexEscape(*text), "\"");`

 (from "absl/strings/escaping.h")

**Comment by kamahen** [comment]
I think I've incorporated all your changes (I had to do a force-push to github).
My question about StringPrettyPrinter remains open.
(BTW, absl:CHexEscape also escapes "\" even though it's not needed; there doesn't seem to be a way to change this behavior)

**Comment by shahms** [comment]
> I think I've incorporated all your changes (I had to do a force-push to github).
> My question about StringPrettyPrinter remains open.
> (BTW, absl:CHexEscape also escapes "" even though it's not needed; there doesn't seem to be a way to change this behavior)

Thanks! I think slight overzealous escaping is fine. Do you mind running `clang-format -i` over the changed files to silence arc-lint?

**Pull request merged**